app/victoria-metrics/main_test.go

```diff
@@ -0,0 +1,270 @@
+		}
+		resp.Body.Close()
+		return resp.StatusCode == 200
+	}
+	if err := waitFor(testStorageInitTimeout, readyStorageCheckFunc); err != nil {
+		log.Fatalf("http server can't start for %s seconds, err %s", testStorageInitTimeout, err)
+	}
+}
+
+func processFlags() {
+	flag.Parse()
+	for _, fs := range []struct {
+		flag  string
+		value string
+	}{
+		{flag: "storageDataPath", value: storagePath},
+		{flag: "httpListenAddr", value: testHTTPListenAddr},
+		{flag: "graphiteListenAddr", value: testStatsDListenAddr},
+		{flag: "opentsdbListenAddr", value: testOpenTSDBListenAddr},
+		{flag: "loggerLevel", value: "ERROR"},
+	} {
+		// panics if flag doesn't exist
+		if err := flag.Lookup(fs.flag).Value.Set(fs.value); err != nil {
+			log.Fatalf("unable to set %q with value %q, err: %v", fs.flag, fs.value, err)
+		}
+	}
+}
+
+func waitFor(timeout time.Duration, f func() bool) error {
+	fraction := timeout / 10
+	for i := fraction; i < timeout; i += fraction {
+		if f() {
+			return nil
+		}
+	}
+	return fmt.Errorf("timeout")
+}
+
+func tearDown() {
+	vminsert.Stop()
+	vmstorage.Stop()
+	vmselect.Stop()
+	if err := httpserver.Stop(*httpListenAddr); err != nil {
+		log.Fatalf("cannot stop the webservice: %s", err)
+	}
+	os.RemoveAll(storagePath)
+}
+
+func TestInfluxDB(t *testing.T) {
+	t.Parallel()
+	tt := readIn("influxdb", t, fmt.Sprintf("%d", insertionTime.UnixNano()))
+	t.Run("influxdb", func(t *testing.T) {
+		for _, test := range tt {
+			t.Run(test.Name, func(t *testing.T) {
+				t.Parallel()
+				httpWrite(t, "http://127.0.0.1"+testHTTPListenAddr+"/write", test.Data)
+				time.Sleep(testWaitForDataTimeout)
+				data := httpRead(t, "http://127.0.0.1"+testHTTPListenAddr, test.Query)
+				rowContains(t, data, test.Result)
+			})
+		}
+	})
+}
+
+func TestGraphite(t *testing.T) {
+	t.Parallel()
+	tt := readIn("graphite", t, fmt.Sprintf("%d", insertionTime.Unix()))
+	t.Run("graphite", func(t *testing.T) {
+		for _, test := range tt {
+			t.Run(test.Name, func(t *testing.T) {
+				t.Parallel()
+				tcpWrite(t, "127.0.0.1"+testStatsDListenAddr, test.Data)
+				time.Sleep(testWaitForDataTimeout)
+				data := httpRead(t, "http://127.0.0.1"+testHTTPListenAddr, test.Query)
+				rowContains(t, data, test.Result)
+			})
+		}
+	})
+}
+
+func TestOpenTSDB(t *testing.T) {
+	t.Parallel()
+	tt := readIn("opentsdb", t, fmt.Sprintf("%d", insertionTime.Unix()))
+	t.Run("opentsdb", func(t *testing.T) {
+		for _, test := range tt {
+			t.Run(test.Name, func(t *testing.T) {
+				t.Parallel()
+				tcpWrite(t, "127.0.0.1"+testOpenTSDBListenAddr, test.Data)
+				time.Sleep(testWaitForDataTimeout)
+				data := httpRead(t, "http://127.0.0.1"+testHTTPListenAddr, test.Query)
+				rowContains(t, data, test.Result)
+			})
+		}
+	})
+}
+
+func readIn(readFor string, t *testing.T, timeStr string) []test {
+	t.Helper()
+	s := newSuite(t)
+	var tt []test
+	s.noError(filepath.Walk(filepath.Join(testFixturesDir, readFor), func(path string, info os.FileInfo, err error) error {
```

Comment:
It doesn't look like `WalkFunc` can return any error since you already "wrap" all of them already. I'd suggest to not wrap results with `s.noError` inside of this function